tests/test-cases/Group23-VIC-Machine-Service/23-00-Util.robot

```diff
@@ -0,0 +1,63 @@
+# Copyright 2017 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+*** Settings ***
+Documentation	This resource contains keywords which are helpful for using curl to test the vic-machine API.
+
+
+*** Variables ***
+${HTTP_PORT}	1337
+${HTTPS_PORT}	31337
+
+${RC}		The return code of the last curl invocation
+${OUTPUT}	The output of the last curl invocation
+${STATUS}	The HTTP status of the last curl invocation
+
+
+*** Keywords ***
+
+Start VIC Machine Server
+    Start Process    ./bin/vic-machine-server --port ${HTTP_PORT} --scheme http    shell=True    cwd=/go/src/github.com/vmware/vic
+
+
+Get Path
+    [Arguments]    ${path}
+    ${RC}  ${OUTPUT}=    Run And Return Rc And Output    curl -s -w "\n\%{http_code}\n" -X GET "http://127.0.0.1:${HTTP_PORT}/container/${PATH}"
+    ${STATUS}=    Get Line    ${OUTPUT}    -1
+    Set Test Variable    ${RC}
+    Set Test Variable    ${OUTPUT}
+    Set Test Variable    ${STATUS}
+
+
+Get Path Under Target
+    [Arguments]    ${path}
+    ${auth}=    Evaluate    base64.b64encode("%{TEST_USERNAME}:%{TEST_PASSWORD}")    modules=base64
+    ${RC}  ${OUTPUT}=    Run And Return Rc And Output    curl -s -w "\n\%{http_code}\n" -X GET "http://127.0.0.1:${HTTP_PORT}/container/target/%{TEST_URL}/${PATH}?thumbprint=%{TEST_THUMBPRINT}" -H "Accept: application/json" -H "Authorization: Basic ${auth}"
+    ${STATUS}=    Get Line    ${OUTPUT}    -1
+    Set Test Variable    ${RC}
```

Comment:
I wouldn't suggest having these keywords in a separate file as I think it will be run as test suite as its under test suite dir but @mhagen-vmware can confirm that. Also, I don't like the corresponding `.md` empty file.

Reply:
> I wouldn't suggest having these keywords in a separate file as I think it will be run as test suite as its under test suite dir but @mhagen-vmware can confirm that.

If it gets run as a suite, it seems like it will get run as a suite with zero tests. That isn't a problem locally. Is it a problem for our CI infrastructure?

(Replying to @mhagen-vmware here, to keep related conversation in a single thread.)
>  this should be a resource file within our resources folder and needs to be added into resources/Util.robot please

These keywords aren't intended to be used by other test groups. Defining them in the resources folder and adding that file to `resources/Util.robot` essentially adds them to a global namespace.

Do we have a better pattern for sharing keywords between suites within a single test group?

Is there a functional problem with this pattern, or is the complaint only that it's stylistically different than what we've done in the past?

> I don't like the corresponding `.md` file.

I don't like it either, but was attempting to following the formatting conventions. I'd rather use that file as documentation for these keywords, but I think I'd need to format it differently. Are we okay with that?

Reply:
> Is it a problem for our CI infrastructure?

No, but that pattern (having util files under tests) doesn't seem right to me.

> Are we okay with that?

I would suggest removing it for now as we don't have any documentation for existing keywords.

Reply:
> No, but that pattern (having util files under tests) doesn't seem right to me.

I moved the resource file so that it's under resources, and named it in a way that makes it clear it's intended for use within this test group.

> I would suggest removing it for now as we don't have any documentation for existing keywords.

I removed the corresponding md file.